src/System.Memory/tests/ReadOnlySpan/BinarySearch.cs

```diff
@@ -0,0 +1,190 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class ReadOnlySpanTests
+    {
+        public static readonly TheoryData<(uint[] Array, uint Value, int ExpectedIndex)> UIntCases =
+            new TheoryData<(uint[] Array, uint Value, int ExpectedIndex)> {
+                (new uint[] { }, 0u, -1),
+                (new uint[] { 1u }, 0u, -1),
+                (new uint[] { 1u }, 1u, 0),
+                (new uint[] { 1u }, 2u, -2),
+                (new uint[] { 1u, 2u, 4u, 5u }, 0u, -1),
+                (new uint[] { 1u, 2u, 4u, 5u }, 1u, 0),
+                (new uint[] { 1u, 2u, 4u, 5u }, 2u, 1),
+                (new uint[] { 1u, 2u, 4u, 5u }, 3u, -3),
+                (new uint[] { 1u, 2u, 4u, 5u }, 4u, 2),
+                (new uint[] { 1u, 2u, 4u, 5u }, 5u, 3),
+                (new uint[] { 1u, 2u, 4u, 5u }, 6u, -5),
```

Comment:
What about the lookup value existing in multiple indices within the span?

Reply:
Also, I would consider adding a test where the lookup value only exists 1 index before/after the span being searched (and hence return -1, i.e. not found) to increase confidence that there aren't any off-by-one errors in our implementation.